### PR TITLE
cloud: MSP API deprecation

### PR DESCRIPTION
**Pull request by Oreoxmt**
### What is changed, added or deleted? (Required)

MSP API is deprecated.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v8.4 (TiDB 8.4 versions)
- [ ] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by yitso — APPROVED** [review]

File: tidb-cloud/api-overview.md [hunk 1]
```diff
@@ -27,6 +27,6 @@ To start using TiDB Cloud API, refer to the following resources in TiDB Cloud AP
         - [Billing](https://docs.pingcap.com/tidbcloud/api/v1beta1/billing)
         - [Data Service](https://docs.pingcap.com/tidbcloud/api/v1beta1/dataservice)
         - [IAM](https://docs.pingcap.com/tidbcloud/api/v1beta1/iam)
-        - [MSP](https://docs.pingcap.com/tidbcloud/api/v1beta1/msp)
+        - [MSP (deprecated)](https://docs.pingcap.com/tidbcloud/api/v1beta1/msp)
```
Comment (qiancai) [inline tidb-cloud/api-overview.md:30]:
```suggestion
        - [MSP (Deprecated)](https://docs.pingcap.com/tidbcloud/api/v1beta1/msp)
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

**Pull request merged**